### PR TITLE
feat: [PIE-3444]: Integrate cypress image snapshot in NG

### PR DESCRIPTION
**Pull request by ishantmehtaharness**
##### Screenshots:

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Comment by ishantmehtaharness** [comment]
retrigger titlecheck

File: cypress/support/commands.ts [hunk 1]
```diff
@@ -78,10 +79,30 @@ declare global {
       verifyStepSelectConnector(): void
       verifyStepChooseRuntimeInput(): void
       verifyStepSelectStrategyAndVerifyStep(): void
+      // https://github.com/jaredpalmer/cypress-image-snapshot
+      matchImageSnapshot(snapshotName?: string, options?: any): void
```
Comment (diptiman-adak) [inline cypress/support/commands.ts:83]: Codacy has a fix for the issue: [Unexpected any. Specify a different type.](https://app.codacy.com/gh/harness/harness-core-ui/pullRequest?prid=9232271)
```suggestion
      matchImageSnapshot(snapshotName?: string, options?: unknown): void
```

**Comment by ishantmehtaharness** [comment]
retrigger jest

File: cypress/support/commands.ts [hunk 2]
```diff
@@ -78,10 +79,30 @@ declare global {
...
+      // https://github.com/jaredpalmer/cypress-image-snapshot
+      matchImageSnapshot(snapshotName?: string, options?: unknown): void
     }
   }
 }
 
+// We set up the settings
+addMatchImageSnapshotCommand({
+  customSnapshotsDir: 'cypress/snapshots',
+  failureThreshold: 2, // threshold for entire image
```
Comment (ishantmehtaharness) [inline cypress/support/commands.ts:91]: We are adding more snapshots in 70-pipeline, and this number will be changed to something lesser than 1% in following PR's

**Review by kapilkumawat86 — APPROVED** [review]

**Review by AshwinBhatkal-Harness — APPROVED** [review]

**Pull request merged**